Library/ViewModels/PledgeShippingLocationViewModel.swift

```diff
@@ -52,17 +52,44 @@ public final class PledgeShippingLocationViewModel: PledgeShippingLocationViewMo
     let selectedLocationId = configData
       .map { $0.3 }
 
-    let shippingShouldBeginLoading = reward
-      .map { $0.shipping.enabled }
-
-    let shippingRulesEvent = Signal.zip(project, reward)
-      .filter { _, reward in reward.shipping.enabled }
-      .switchMap { project, reward -> SignalProducer<Signal<[ShippingRule], ErrorEnvelope>.Event, Never> in
-        AppEnvironment.current.apiService.fetchRewardShippingRules(projectId: project.id, rewardId: reward.id)
+    let shippingShouldBeginLoading = configData
+      .mapConst(true)
+    
+    // CURRENT APP VERSION:
+      
+//    let shippingRulesEvent = Signal.zip(project, reward)
+//      .filter { _, reward in reward.shipping.enabled }
+//      .switchMap { project, reward -> SignalProducer<Signal<[ShippingRule], ErrorEnvelope>.Event, Never> in
+//        AppEnvironment.current.apiService.fetchRewardShippingRules(projectId: project.id, rewardId: reward.id)
+//          .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+//          .map(ShippingRulesEnvelope.lens.shippingRules.view)
+//          .retry(upTo: 3)
+//          .materialize()
+//      }
+
+    let shippingRulesEvent = project
+      .switchMap { project -> SignalProducer<Signal<[ShippingRule], ErrorEnvelope>.Event, Never> in
+        /// Get  all of the reward IDs we'll need to fetch the Shipping Rules. See inner method logic for more details.
+        let rewardIDsToQuery: Set<Int> = getRewardIDsToQuery(for: project)
+
+        /// Initializing the result with a fetch using the first reward ID in the Set to avoid optional warnings
+        var queryResult: SignalProducer<Signal<[ShippingRule], ErrorEnvelope>.Event, Never>?
+
+        /// Ffetch the shipping rules for each reward and then consolidate each corresponding SignalProducer into the `queryResult` variable using .merge(with:).
+        rewardIDsToQuery.forEach { id in
+          let fetchResult = AppEnvironment.current.apiService.fetchRewardShippingRules(
+            projectId: project.id,
+            rewardId: id
+          )
           .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
           .map(ShippingRulesEnvelope.lens.shippingRules.view)
           .retry(upTo: 3)
           .materialize()
+
+          queryResult = queryResult?.merge(with: fetchResult) ?? fetchResult
+        }
+
+        return queryResult ?? SignalProducer(value: .value([]))
```

Comment:
This solution of grabbing the IDs of the rewards that will help us determine what Shipping Rules to get, then looping through and calling the V1 endpoint for each one makes sense in theory, but I'm wondering if this is the best RxSwift pattern.

I would definitely love some help refactoring this if possible.

Reply:
I need a way to capture each returned SignalProducer from each `fetchRewardShippingRules` call and then essentially combine them into one resulting `Signal<[ShippingRule], ErrorEnvelope>` hence the `.merge(with:)` on line: 89.

I'm not sure that this is very RxSwifty. It requires `queryResult` to be `nil` which  _might_ be ok since there shouldn't be a case where the project has no rewards. If the `rewardIDsToQuery` is empty, it would likely break things.